tests/test_election.py

```diff
@@ -0,0 +1,133 @@
+"""
+Unit Tests for Election Models
+
+These tests may fail at any time due to the whims of the Blaseball API. See also: Blaseball Falsehoods
+"""
+
+import pytest
+from blaseball_mike.models import SimulationData, ElectionResult, Election, DecreeResult,\
+    BlessingResult, TidingResult, Team
+from tests.test_base import base_test
+
+
+def pytest_generate_tests(metafunc):
```

Comment:
while neat, I think using `pytest_generate_tests` is going to add a bunch of maintenance overhead given all the introspection, and the fact that it dynamically parametrizes using using the live API adds instability and uncertainty. Even after using VCR to lock the HTTP response, it's still going to be difficult to debug the input if something weird happens. Also `pytest_generate_tests` is like really poorly documented.

My preference is to define static fixtures in `conftest`. You can even use the same instantiation logic so it's not a complete rewrite. Even though that doesn't fully solve the second point, at the very least it makes the fixtures reusable and a bit easier to copy/edit as the new cases arise going forward.

One other thing that this shotgun approach is causing a 10 minute test run. This library is not nearly large or complex enough to warrant a 10 minute test run. Part of this is definitely the HTTP calls, but also I suspect that many of the test cases that get automagically loaded aren't testing any new code paths.

Reply:
I agree, particularly after looking into VCR more. Implementing it with the existing collection steps is... not fun. 

The main reason I built it like this is to verify that the load functions return what we expect, which has already found a number of issues (one of which I just found yesterday with Player.load_by_gameday) but you're right that we really only need one for each load function and that can probably just be in a separate fixture-ized test function for each load type.

And I didn't actually know about `conftest.py` until doing some digging this morning. pytest sure seems to like having features defined by poorly defined file/function names...